amplipi/streams.py

```diff
@@ -1073,25 +1079,56 @@ def connect(self, src):
           # NOTE: port 9000 is assumed
           server.replace('localhost', socket.gethostname())
         lms_args += [ '-s', server]
+
+        meta_args = ['python3', 'streams/lms_metadata.py', '--name', self.name, "--client", self.server]
+        self.meta_proc = subprocess.Popen(args=meta_args)
+      else:
+        meta_args = ['python3', 'streams/lms_metadata.py', '--name', self.name]
+        self.meta_proc = subprocess.Popen(args=meta_args)
       # TODO: allow port to be specified with server (embedding it in the server URL does not work)
 
       self.proc = subprocess.Popen(args=lms_args)
       self._connect(src)
+
+      # TODO: Add metadata support? This may have to watch the output log?
+      # At the end of the connect function because the LMS stream needs to be fully initialized before it starts searching for metadata or you won't be able to connect the player
+
+
     except Exception as exc:
       print(f'error starting lms: {exc}')
 
   def disconnect(self):
     if self._is_running():
       self.proc.kill()
     self._disconnect()
+    if self.meta_proc is not None:
+      self.meta_proc.terminate()
+      self.meta_proc = None
     self.proc = None
+    if os.path.exists(f"lms_{str(self.name).replace(' ', '_')}_metadata.json"):
+      os.remove(f"lms_{str(self.name).replace(' ', '_')}_metadata.json")
 
   def info(self) -> models.SourceInfo:
+    # Opens and reads the metadata.json file every time the info def is called
+    # uses a file lock so that the file cannot be read while writing and vice/versa. If you let those processes run into eachother, there are errors
+    try:
+      with open(f"lms_{str(self.name).replace(' ', '_')}_metadata.json", "r", encoding="utf-8") as meta_read:
+        self.meta = json.loads(meta_read.read())
+    except:
+      self.meta = {
+        'track': 'Trying again shortly...',
+        'album': 'Make sure your lms player is connected to this source',
+        'artist': 'Error: Could Not Find LMS Server',
+        'image_url': 'static/imgs/lms.png'
+      }
+
     source = models.SourceInfo(
       name=self.full_name(),
       state=self.state,
-      img_url='static/imgs/lms.png',
-      track='check LMS for song info',
+      img_url= self.meta['image_url'],
```

Comment:
there is no garauntee that all of those fields exist. What if album is not populated into self.meta?

Reply:
How about something like `self.meta.get('track', '')`